include/boost/gil/extension/numeric/convolve.hpp

```diff
@@ -169,6 +169,16 @@ void convolve_cols(const SrcView& src, const Kernel& ker, const DstView& dst,
     convolve_rows<PixelAccum>(transposed_view(src),ker,transposed_view(dst),option);
 }
 
+/// \ingroup ImageAlgorithms
+///convolve a 1D variable-size kernel along both rows and columns of an image
+template <typename PixelAccum,typename SrcView,typename Kernel,typename DstView>
+BOOST_FORCEINLINE
+void convolve(const SrcView& src, const Kernel& ker, const DstView& dst,
+                   convolve_boundary_option option=convolve_option_extend_zero) {
+    convolve_rows<PixelAccum>(src, ker, dst, option);
+    convolve_cols<PixelAccum>(dst, ker, dst, option);
```

Comment:
This source and destination views overlap in the second call makes me suspicious.
I always tend to use that with intermediate buffer.
On the other hand, the `convolve_rows/cols` implementation does the buffering.